utils/ips/ip_port.go

```diff
@@ -12,7 +12,44 @@ import (
 	"github.com/ava-labs/avalanchego/utils/wrappers"
 )
 
-var errBadIP = errors.New("bad ip format")
+const nullStr = "null"
+
+var (
+	errMissingQuotes = errors.New("first and last characters should be quotes")
+	errBadIP         = errors.New("bad ip format")
+)
+
+type IPDesc (IPPort)
+
+func (ipDesc IPDesc) String() string {
+	return IPPort(ipDesc).String()
+}
+
+func (ipDesc IPDesc) MarshalJSON() ([]byte, error) {
+	return []byte("\"" + fmt.Sprintf("%s:%d", ipDesc.IP, ipDesc.Port) + "\""), nil
+}
+
+func (ipDesc *IPDesc) UnmarshalJSON(b []byte) error {
+	str := string(b)
+	if str == nullStr { // If "null", do nothing
```

Comment:
I'm just confused about this edge-case, why do we need this check? Does empty string not work in place of "null"?

Reply:
https://pkg.go.dev/encoding/json#Unmarshaler

> Unmarshaler is the interface implemented by types that can unmarshal a JSON description of themselves. The input can be assumed to be a valid encoding of a JSON value. UnmarshalJSON must copy the JSON data if it wishes to retain the data after returning.
> 
> By convention, to approximate the behavior of Unmarshal itself, Unmarshalers implement UnmarshalJSON([]byte("null")) as a no-op.